Mutagen.Bethesda.Analyzers.Engine/Reporting/Drops/IReportDropbox.cs

```diff
@@ -0,0 +1,17 @@
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins.Records;
+
+namespace Mutagen.Bethesda.Analyzers.Reporting.Drops;
+
+public interface IReportDropbox
```

Comment:
Somehow feels a bit strange to have an exact copy of an interface for this with `IReportHandler`. I this is needed for proper DI handling looking at something like `PassToHandlerReportDropbox` which needs to get all the `IReportHandler` implementations injected but it feels a bit like a technical necessity instead of explicit design? I can't quite think of a better way to handle this right now though. There could be one common base interface that both are extending to avoid the duplicate Dropoff definitions but that only works if we're sure the definitions are always going to be coupled between `IReportHandler` and `IReportDropbox`.

Reply:
Yeah, right now it's because the interfaces are handled differently by the DI.   IReportDropbox is supposed to be a decorator pipeline where there's one but it wraps the others in a conveyor belt.  But then it shifts to a "broadcast" style for IReportHandler[] where we want to get all of the handlers that exist and loop/report all of them.

Not sure how to have that transition from decorator pipeline to broadcast while using the same interface.   We can definitely evolve it as we go, as it's all internal mechanisms to the engine